tasks/testwasher.py

```diff
@@ -45,10 +45,12 @@ def get_non_flaky_failing_tests(self, module_path):
         for package, tests in failing_tests.items():
             non_flaky_failing_tests_in_package = set()
             for failing_test in tests:
-                if all(
-                    not failing_test.startswith(flaky_marked_test) for flaky_marked_test in flaky_marked_tests[package]
-                ) and all(
-                    not failing_test.startswith(flaky_marked_test)
+                failing_test_parent = failing_test.split("/")[0]
+                if not any(
+                    flaky_marked_test.startswith(failing_test_parent)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You can keep this operation 0(1) by extracting all the parents from the failing test and then looping through all parents. I am not sure this would work if you have 3 + level of nested tests. For example:

TestName: `"TestEKSSuite/TestCPU/metric___container.cpu.usage{^kube_deployment:stress-ng$,^kube_namespace:workload-cpustress$}`
KnownFlake: `TestEKSSuite`

Here it looks like you only check for its direct parent `TestEKSSuite/TestCPU`, so it would consider as an unexpected error failures of `"TestEKSSuite/TestCPU/metric___container.cpu.usage{^kube_deployment:stress-ng$,^kube_namespace:workload-cpustress$}` giving `TestEKSSuite` is marked.

What I would do it instead extract a function `is_known_flake` passing test name, test package and `flaky_marked_test`

Then the function would be

```python
def get_test_family(test_name):
    test_family = [test_name]
    while test_name.count('/') > 0:
        test_name = test_name.rsplit('/',1)[0]
        test_family.append(test_name)
    return test_family	

def is_known_flaky_tests(self, test_package, test_name, flaky_marked_tests):
    # early return if package is unknown
    if empty(flaky_marked_test[package]) and empty(self.known_flaky_tests[package]):
      return false
	# extract test and its parents
    test_family = get_test_family(test_name)
    
    for  test in test_family:
		if test in flaky_marked_test[package] OR test in self.known_flaky_tests[package]:
			return true

	return false
```

Example: https://www.online-python.com/3UEgHZPY9O

Reply:
Also the marking for this specific case should be in the root `flakes.yaml` through the following content

```yaml
"github.com/DataDog/datadog-agent/test/new-e2e/tests/containers":
  - "TestEKSSuite/TestCPU/metric___container.cpu.usage{^kube_deployment:stress-ng$,^kube_namespace:workload-cpustress$}"
```

As `TestCPU` is defined in `k8sSuite`, all Suites embedding `k8sSuite` would be marked as well, if `TestKindSuite/TestCPU` is not flaky we should not mark it.